GhidraBuild/IDAPro/Python/7xx/python/idaxml.py

```diff
@@ -358,16 +360,19 @@ def check_char(self, ch):
             String containing either the character or the entity
             substition string.
         """
-        if ((ord(ch) < 0x20) and (ord(ch) != 0x09 and
-             ord(ch) != 0x0A and ord(ch) != 0x0D)): return ''
+        x = ch
+        if not isinstance(ch, int):
+            x = ord(ch)
+        if ((x < 0x20) and (x != 0x09 and
+             x != 0x0A and x != 0x0D)): return ''
         elif ch == '&' :  return '&amp;'
```

Comment:
@ikelos I don't see why as the variable ch is the actual character. It is not the "x"

Reply:
Sorry, you're right, I meant `x` not `ch`.

The new line 364 suggests `ch` may *not* always be the actual character, it may now be an integer.  If it's an integer, the following tests against `ch` will all fail and produce invalid XML.

To fix this *all* the tests should only happen against the normalized integer `x` instead of the duck-typed could-be-an-int-could-be-a-str value `ch`.

Reply:
please see my new commit.

Reply:
Yep, I think that sorts it, thanks.